packs/tfe_fdo_nomad/templates/tfe.nomad.tpl

```diff
@@ -0,0 +1,220 @@
+        env         = true
+        change_mode = "restart"
+        data        = <<EOF
+{{- with nomadVar "nomad/jobs/[[ .tfe_fdo_nomad.job_name ]]" -}}
+TFE_LICENSE                             = {{ .tfe_license }}
+TFE_DATABASE_PASSWORD                   = {{ .db_password }}
+TFE_OBJECT_STORAGE_S3_SECRET_ACCESS_KEY = {{ .s3_secret_key }}
+TFE_REDIS_PASSWORD                      = {{ .redis_password }}
+TFE_ENCRYPTION_PASSWORD                 = {{ .tfe_encryption_password }}
+TFE_IMAGE_REGISTRY_PASSWORD             = {{ .tfe_image_registry_password }}
+{{- end -}}
+EOF
+      }
+
+      config {
+         image = [[ .tfe_fdo_nomad.tfe_image | quote ]]
+         auth {
+            # User Input is mandatory.
+            username = [[ .tfe_fdo_nomad.tfe_image_registry_username | quote ]]
+            # User Input is mandatory.
+            password       = "${TFE_IMAGE_REGISTRY_PASSWORD}"
+            server_address = [[ .tfe_fdo_nomad.tfe_image_server_address | quote ]]
+         }
+         ports = ["tfe", "http", "vault"]
+
+         volumes = [
+          "secrets:[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]",
+        ]
+  
+      }
+
+      env {
+        
+        TFE_RUN_PIPELINE_DRIVER = "nomad"
+        # User Input is mandatory. 
+        TFE_RUN_PIPELINE_NOMAD_ADDRESS             = [[ .tfe_fdo_nomad.tfe_run_pipeline_nomad_address | quote ]]
+        TFE_RUN_PIPELINE_NOMAD_TLS_CONFIG_INSECURE = [[ .tfe_fdo_nomad.tfe_run_pipeline_nomad_tls_config_insecure | quote ]]
+        TFE_RUN_PIPELINE_NOMAD_TLS_CONFIG_CA_CERT = "[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]/nomad_ca_cert.pem"
+        TFE_RUN_PIPELINE_NOMAD_TLS_CONFIG_CLIENT_CERT = "[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]/nomad_cert.pem"
+        TFE_RUN_PIPELINE_NOMAD_TLS_CONFIG_CLIENT_KEY = "[[ .tfe_fdo_nomad.tfe_tls_cert_mount_path ]]/nomad_cert_key.pem"
+        TFE_DISK_CACHE_VOLUME_NAME                 = "${NOMAD_TASK_DIR}/terraform-enterprise-cache-1"
+
+        TFE_OPERATIONAL_MODE = "active-active"
+
+        TFE_DATABASE_USER = [[ .tfe_fdo_nomad.tfe_database_user | quote ]]
+        # User Input is mandatory. 
+        TFE_DATABASE_HOST       = [[ .tfe_fdo_nomad.tfe_database_host | quote ]]
+        TFE_DATABASE_NAME       = [[ .tfe_fdo_nomad.tfe_database_name | quote ]]
+        TFE_DATABASE_PARAMETERS = [[ .tfe_fdo_nomad.tfe_database_parameters | quote ]]
+
+        TFE_OBJECT_STORAGE_TYPE = [[ .tfe_fdo_nomad.tfe_object_storage_type | quote ]]
+        TFE_OBJECT_STORAGE_S3_BUCKET               = [[ .tfe_fdo_nomad.tfe_object_storage_s3_bucket | quote ]]
+        TFE_OBJECT_STORAGE_S3_REGION               = [[ .tfe_fdo_nomad.tfe_object_storage_s3_region | quote ]]
+        TFE_OBJECT_STORAGE_S3_USE_INSTANCE_PROFILE = [[ .tfe_fdo_nomad.tfe_object_storage_s3_use_instance_profile ]]
+        TFE_OBJECT_STORAGE_S3_ENDPOINT             = [[ .tfe_fdo_nomad.tfe_object_storage_s3_endpoint | quote ]]
+        # User Input is mandatory. 
```

Comment:
What do these repeated comments `# User Input is mandatory` mean? If they're meant to say that the Pack variables are required, that's already controlled by the variables definition file.